python/taulabs/telemetry.py

```diff
@@ -323,13 +323,19 @@ def _do_io(self, finish_time):
         if r:
             # Shouldn't throw an exception-- they just told us
             # it was ready for read.
-            chunk = os.read(self.fd, 1024)
-            if chunk == '':
-                raise RuntimeError("stream closed")
+            try:
```

Comment:
@peabody124 no, the simulator works normal and successive read calls are successful. I can add a note to the code.